fast_llm/layers/language_model/head.py

```diff
@@ -436,14 +429,83 @@ def _logits_cross_entropy_forward_backward(
         loss = _add_tensors(dpo_loss, lm_loss, distillation_loss)
         if self.training and losses is not None:
             if dpo_loss is not None:
-                losses[LanguageModelLossNames.dpo_loss].append(dpo_loss.detach())
+                losses[self._dpo_loss_name].append(dpo_loss.detach())
             if self._config.distillation_model is not None and distillation_loss is not None:
-                losses[LanguageModelLossNames.distillation_loss].append(distillation_loss.detach())
+                losses[self._distillation_loss_name].append(distillation_loss.detach())
             if self._config.distillation_model is not None and lm_loss is not None:
-                losses[LanguageModelLossNames.distil_lm_loss].append(lm_loss.detach())
+                losses[self._distillation_language_model_loss_name].append(lm_loss.detach())
```

Comment:
it would help to clarify the purposes of the 1) detached losses, 2) loss definitions, and 3) non-detached returned loss.
1) and 2) are 1-on-1 related, but idk if 3) and 2) are.